### PR TITLE
Remove all direct usages of slf4j / forward slf4j to log4j

### PR DESCRIPTION
**Pull request by jjohannes**
**Description**:

Some SLF4J usages sneaked in through transitive dependencies I suppose. In the dependency "cleanup" (e.g. #7875) this was not yet addressed. Instead now you sometimes see:

```
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See https://www.slf4j.org/codes.html#noProviders for further details.
```

This problem was suppressed [before](https://github.com/hashgraph/hedera-services/blob/3893445245b2c290beafe776ea26cb414c8fc53f/platform-sdk/settings.gradle.kts#L143) by adding `slf4j-nop` which swallowed the logging. This is already gone on develop (hence the message above).

This PR does two things:
1. Exchange all usages of "slf4j" in the code with "log4j"
2. Add the `org.apache.logging.log4j.slf4` bridge as a central "runtime only" dependency that is added to the runtime classpath of all tests/applications. This forwards the slf4j logs to log4j.

**Related issue(s)**:

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by tinker-michaelj — DISMISSED** [review]
LGTM!

File: build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts [hunk 1]
```diff
@@ -54,9 +54,19 @@ val internal: Configuration =
...
+    dependencies {
+        // For dependencies of annotation processors use versions from 'hedera-dependency-versions',
+        // but not 'runtime' dependencies of the platform (JAVA_API instead of JAVA_RUNTIME).
+        annotationProcessorConfigurationName("com.hedera.hashgraph:hedera-dependency-versions") {
+            attributes {
+                attribute(Usage.USAGE_ATTRIBUTE, objects.named(Usage.JAVA_API))
+                attribute(Category.CATEGORY_ATTRIBUTE, objects.named(Category.REGULAR_PLATFORM))
+            }
+        }
+    }
```
Comment (jjohannes) [inline build-logic/project-plugins/src/main/kotlin/com.hedera.hashgraph.java.gradle.kts:69]: These adjustments for how the "dependency platform" (hedera-dependency-versions) is used for annotation processing and protobuf generation are necessary as we not also use it as a central place to define runtime only dependencies for test/application execution.
We do not want those dependencies on the path of the annotation processors or protobufs generator.

File: hedera-dependency-versions/build.gradle.kts [hunk 2]
```diff
@@ -26,7 +26,7 @@ val eclipseCollectionsVersion = "10.4.0"
 val grpcVersion = "1.54.1"
 val helidonVersion = "3.2.1"
 val jacksonVersion = "2.13.5"
-val log4jVersion = "2.17.2"
+val log4jVersion = "2.20.0"
```
Comment (jjohannes) [inline hedera-dependency-versions/build.gradle.kts:29]: Update necessary to for SLF4 **2** integration

File: hedera-dependency-versions/build.gradle.kts [hunk 3]
```diff
@@ -37,6 +37,9 @@ val tuweniVersion = "2.3.1"
 
 dependencies {
     api(enforcedPlatform("io.netty:netty-bom:$nettyVersion"))
+
+    // forward logging from modules using SLF4J (e.g. 'org.hyperledger.besu.evm') to Log4J
+    runtime(javaModuleDependencies.gav("org.apache.logging.log4j.slf4j"))
```
Comment (jjohannes) [inline hedera-dependency-versions/build.gradle.kts:42]: Central place to define additional dependencies we want for all "runtimes" (test execution / application execution) but do not want to publish. If our modules are used somewhere else, the consumer should be able to decide on logger implementation(s). (Ideally, log4jcore could also move here such that our code only depends on log4j-api.)

File: hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/util/LogCaptor.java [hunk 4]
```diff
@@ -114,7 +114,7 @@ private List<String> eventsAt(@NonNull final Level level) {
             }
             // now check if the current match is the level we're looking for
             String matchLevel = m.group(0);
-            if (level.equals(Level.getLevel(matchLevel))) {
+            if (!matchLevel.isEmpty() && level.equals(Level.getLevel(matchLevel))) {
```
Comment (jjohannes) [inline hedera-node/hedera-app-spi/src/testFixtures/java/com/hedera/node/app/spi/fixtures/util/LogCaptor.java:117]: Needed due to the LOG4J update. `getLevel` now throws in an empty string is passed.

**Pull request merged**